App crashes when going outside of range of hotspot, or having it turned off

**To Reproduce**
Steps to reproduce the behavior:
1. Turn on the hotspot
2. Connect the main robot phone to the hotspot
3. Turn on the robot, and wait for the video to stabilize
4. Turn hotspot off

**Expected behavior**
App will crash


**Smartphone (please complete the following information):**
 - Device: Samsung Galaxy S4
 - OS: Android 5.0
 - App Version v0.9.3-beta

**Additional context**
This might be from it switching back to a known WiFi, but not sure yet. Having issues getting the logs to output the reason for the crash other than FFmpeg failing, which does not produce crashes on its own

```
tv.remo.android.controller E/AndroidRuntime: FATAL EXCEPTION: AudioComponent
    Process: tv.remo.android.controller, PID: 15319
    java.io.IOException: write failed: EPIPE (Broken pipe)
        at libcore.io.IoBridge.write(IoBridge.java:502)
        at java.io.FileOutputStream.write(FileOutputStream.java:186)
        at java.io.OutputStream.write(OutputStream.java:82)
        at org.btelman.controlsdk.streaming.utils.OutputStreamUtil.handleSendByteArray(OutputStreamUtil.kt:32)
        at org.btelman.controlsdk.streaming.audio.processors.FFmpegAudioProcessor.processAudioByteArray(FFmpegAudioProcessor.kt:67)
        at org.btelman.controlsdk.streaming.components.AudioComponent.doWorkLoop(AudioComponent.kt:23)
        at org.btelman.controlsdk.streaming.components.StreamComponent.fetchFrame(StreamComponent.kt:47)
        at org.btelman.controlsdk.streaming.components.StreamComponent.handleMessage(StreamComponent.kt:40)
        at org.btelman.controlsdk.models.Component$handler$1.handleMessage(Component.kt:44)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:145)
        at android.os.HandlerThread.run(HandlerThread.java:61)
     Caused by: android.system.ErrnoException: write failed: EPIPE (Broken pipe)
        at libcore.io.Posix.writeBytes(Native Method)
        at libcore.io.Posix.write(Posix.java:223)
        at libcore.io.BlockGuardOs.write(BlockGuardOs.java:313)
        at libcore.io.IoBridge.write(IoBridge.java:497)
        at java.io.FileOutputStream.write(FileOutputStream.java:186) 
        at java.io.OutputStream.write(OutputStream.java:82) 
        at org.btelman.controlsdk.streaming.utils.OutputStreamUtil.handleSendByteArray(OutputStreamUtil.kt:32) 
        at org.btelman.controlsdk.streaming.audio.processors.FFmpegAudioProcessor.processAudioByteArray(FFmpegAudioProcessor.kt:67) 
        at org.btelman.controlsdk.streaming.components.AudioComponent.doWorkLoop(AudioComponent.kt:23) 
        at org.btelman.controlsdk.streaming.components.StreamComponent.fetchFrame(StreamComponent.kt:47) 
        at org.btelman.controlsdk.streaming.components.StreamComponent.handleMessage(StreamComponent.kt:40) 
        at org.btelman.controlsdk.models.Component$handler$1.handleMessage(Component.kt:44) 
        at android.os.Handler.dispatchMessage(Handler.java:98) 
        at android.os.Looper.loop(Looper.java:145) 
        at android.os.HandlerThread.run(HandlerThread.java:61) 
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -20,8 +20,8 @@ android {
         applicationId "tv.remo.android.controller"
         minSdkVersion 16
         targetSdkVersion 28
-        versionCode 4
-        versionName "0.10.0"
+        versionCode 5
+        versionName "0.10.1"
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
     buildTypes {

--- a/sdk/build.gradle
+++ b/sdk/build.gradle
@@ -33,12 +33,13 @@ dependencies {
     androidTestImplementation 'androidx.test:runner:1.2.0'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.2.0'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    def controlsdk_version = "0.8.1"
+    def controlsdk_version = "0.8.2"
     api "org.btelman.controlsdk:core:$controlsdk_version"
     api "org.btelman.controlsdk:streaming:$controlsdk_version"
     api "org.btelman.controlsdk:hardware:$controlsdk_version"
     api "org.btelman.controlsdk:texttospeech:$controlsdk_version"
-    api 'com.squareup.okhttp3:okhttp:3.12.+'
+    //noinspection GradleDependency. We cannot upgrade this without losing support for Android 4.4 and below
+    api 'com.squareup.okhttp3:okhttp:3.12.6'
     implementation 'com.google.code.gson:gson:2.8.5'
     api project(path: ':licensehelper')
 }

--- a/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
+++ b/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
@@ -150,7 +150,7 @@ class RemoSocketComponent : Component() {
     private fun verifyAndSubToChannel(json: String) {
         serverInfo = Gson().fromJson(json, RobotServerInfo::class.java).also { serverInfo ->
             for (channel in serverInfo.channels) {
-                if(channel.id != activeChannelId) continue
+                if(channel.id != activeChannelId && channel.name != activeChannelId) continue
                 activeChannel = channel
                 sendChatMessage("Robot connected. Commands cleared")
             }

--- a/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
+++ b/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
@@ -97,7 +97,7 @@ class RemoSocketComponent : Component() {
     private fun sendChatUpwards(json: String) {
         Gson().fromJson(json, Message::class.java).also {
             if(it.type == "robot") return //we don't want the robot talking to itself
-            if(activeChannel?.chat != it.chatId) return
+            if(activeChannel?.id != it.channelId) return
             if(searchAndSendCommand(it)) return
             val data = TTSBaseComponent.TTSObject(
                 it.message,

--- a/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
+++ b/sdk/src/main/java/tv/remo/android/controller/sdk/components/RemoSocketComponent.kt
@@ -28,7 +28,9 @@ class RemoSocketComponent : Component() {
     private var socket: WebSocket? = null
     var apiKey : String? = null
     var activeChannelId : String? = null
-    lateinit var request : Request
+    private val listener = SocketListener()
+    private var url: String? = null
+    var request : Request? = null
     val client = OkHttpClient()
     private var serverInfo: RobotServerInfo? = null
     private var activeChannel : Channel? = null
@@ -36,16 +38,14 @@ class RemoSocketComponent : Component() {
     override fun onInitializeComponent(applicationContext: Context, bundle: Bundle?) {
         super.onInitializeComponent(applicationContext, bundle)
         apiKey = bundle?.getString(API_TOKEN_BUNDLE_KEY)
-        request = Request.Builder().url(EndpointBuilder.buildWebsocketUrl(applicationContext)).build()
+        url = EndpointBuilder.buildWebsocketUrl(applicationContext)
         activeChannelId = bundle?.getString(CHANNEL_ID_BUNDLE_KEY)
         apiKey?: throw Exception("api key not found")
     }
 
     override fun enableInternal() {
-        val listener = SocketListener()
         subToSocketEvents(listener)
-        socket = client.newWebSocket(request, listener)
-        client.dispatcher().executorService().shutdown()
+        attemptReconnect()
     }
 
     override fun disableInternal() {
@@ -69,6 +69,10 @@ class RemoSocketComponent : Component() {
         }.on(SocketListener.ON_CLOSE){
             Log.d("TAG","onClosing $it")
         }.on(SocketListener.ON_ERROR){
+            handler.postDelayed ({
+                //attempt a reconnect every second
+                attemptReconnect()
+            }, 1000)
             Log.d("TAG","onFailure $it")
         }.on("ROBOT_VALIDATED"){
             sendChannelsRequest(it)
@@ -82,6 +86,14 @@ class RemoSocketComponent : Component() {
             sendCommandUpwards(it)
         }
         //.on("SEND_CHAT") //TODO? of type Messages
+    }
+
+    private fun attemptReconnect() {
+        url ?: return
+        socket?.close(1000, "service ended normally")
+        request = Request.Builder().url(url!!).build()
+        client.connectionPool().evictAll()
+        socket = client.newWebSocket(request!!, listener)
     }
 
     private fun sendCommandUpwards(it: String) {

--- a/sdk/src/main/java/tv/remo/android/controller/sdk/models/api/Message.kt
+++ b/sdk/src/main/java/tv/remo/android/controller/sdk/models/api/Message.kt
@@ -22,6 +22,8 @@ data class Message(
     val senderId: String,
     @SerializedName("server_id")
     val serverId: String,
+    @SerializedName("channel_id")
+    val channelId: String,
     @SerializedName("time_stamp")
     val timeStamp: Long,
     @SerializedName("type")

